app/jinja_filters.py

```diff
@@ -52,6 +52,29 @@ def format_percentage(value):
     return '{}%'.format(value)
 
 
+@blueprint.app_template_filter()
+def format_address_list(user_entered_address=None, metadata_address=None):
+    """
+        This function format_address_list accepts two addresses, a user submitted address and a metadata address
+        If all the items in address list are 'Undefined' (nothing in answer_store) then we retrieve the metadata.
+
+        :param user_entered_address user entered address which is a list of answer fields
+        :param metadata_address metadata address which is a list of answer fields passed to us
+        :return: the value of the address to be piped
+    """
+    if all(isinstance(field, Undefined) for field in user_entered_address):
+        address = metadata_address
+    else:
+        address = user_entered_address
+
+    address_list = Markup('<br />').join([i for i in address if i])
```

Comment:
Could you use concatenated_list here to keep things consistent?

Reply:
Didn't see that, looks like it works, so gonna test a little 👍

Reply:
Not a big deal really, just a little nicer.